### PR TITLE
Fixed timer to only rerender timer component when time is udpated

### PR DESCRIPTION
**Pull request by emilyseibert**
# PR for Issue #232 .

## PR Status

**READY**

## Changes proposed in the pull request:
- Remove force-update-all which updated every component every second
- Add simple state increment/deref which will only be changed for active timers
- clean up timer intervals when timer runs out or on unmount

## Breaking changes?
- None :)

### CODE REVIEW

File: src/cljs/wombats_web_client/components/countdown_timer.cljs [hunk 1]
```diff
@@ -40,39 +51,46 @@
...
+                                  :timeout-fn
+                                  (.setTimeout js/window
+                                               swap-interval-fn!
+                                               ;; Give the browser some extra time to render the next second
+                                               (- (millis-until start-time) 100))))]
     (reagent/create-class
      {:component-will-mount
       ;; Force timer to redraw every second, start interval when the countdown timer should switch
-      (swap-timeout-fn!)
+      #(swap-timeout-fn!)
```
Comment (dehli) [inline src/cljs/wombats_web_client/components/countdown_timer.cljs]: Why did we turn this into an anonymous function? I don't think it's doing anything like this.

Comment (emilyseibert) [inline src/cljs/wombats_web_client/components/countdown_timer.cljs]: https://github.com/Day8/re-frame/wiki/Creating-Reagent-Components#form-3-a-class-with-life-cycle-methods

so you actually want a function for component will mount

Comment (dehli) [inline src/cljs/wombats_web_client/components/countdown_timer.cljs]: But wouldn't calling `swap-timeout-fn!` result in a function as well?

Comment (emilyseibert) [inline src/cljs/wombats_web_client/components/countdown_timer.cljs]: you're right, but it's working...hmm

Comment (dehli) [inline src/cljs/wombats_web_client/components/countdown_timer.cljs]: 🤔 Let me look through it a little more.

File: src/cljs/wombats_web_client/components/countdown_timer.cljs [hunk 2]
```diff
@@ -40,39 +51,46 @@
...
-      :component-will-unmount
-      (fn []
-        (let [{:keys [interval-fn timeout-fn]} @cmpnt-state]
-          (.clearTimeout js/window timeout-fn)
-          (.clearInterval js/window interval-fn)))
+      :component-will-unmount #(clear-timers cmpnt-state)
 
       :reagent-render
       (fn []
+        ;; triggers a rerender for active timers, will not effect timers with no interval timer
```
Comment (dehli) [inline src/cljs/wombats_web_client/components/countdown_timer.cljs]: typo affect

**Pull request merged**